src/coreclr/interpreter/intrinsics.cpp

```diff
@@ -0,0 +1,121 @@
+#define HAS_PREFIX(haystack, needle) \
+    (strncmp(haystack, needle, strlen(needle)) == 0)
+
+NamedIntrinsic GetNamedIntrinsic(COMP_HANDLE compHnd, CORINFO_METHOD_HANDLE compMethod, CORINFO_METHOD_HANDLE method)
+{
+    const char* className = NULL;
+    const char* namespaceName = NULL;
+    const char* methodName = compHnd->getMethodNameFromMetadata(method, &className, &namespaceName, NULL, 0);
+
+    // Array methods don't have metadata
+    if (!namespaceName)
+        return NI_Illegal;
+
+    if (!HAS_PREFIX(namespaceName, "System"))
+        return NI_Illegal;
+
+    if (!strcmp(namespaceName, "System"))
+    {
+        if (!strcmp(className, "Double") || !strcmp(className, "Single"))
+        {
+            if (!strcmp(methodName, "ConvertToIntegerNative"))
+                return NI_PRIMITIVE_ConvertToIntegerNative;
+            else if (!strcmp(methodName, "MultiplyAddEstimate"))
+                return NI_System_Math_MultiplyAddEstimate;
+        }
+        else if (!strcmp(className, "Math") || !strcmp(className, "MathF"))
+        {
+            if (!strcmp(methodName, "ReciprocalEstimate"))
+                return NI_System_Math_ReciprocalEstimate;
+            else if (!strcmp(methodName, "ReciprocalSqrtEstimate"))
+                return NI_System_Math_ReciprocalSqrtEstimate;
+        }
+    }
+    else if (!strcmp(namespaceName, "System.Numerics"))
+    {
+        if (!strcmp(className, "Vector") && !strcmp(methodName, "get_IsHardwareAccelerated"))
+            return NI_IsSupported_False;
+
+        // Fall back to managed implementation for everything else.
+        return NI_Illegal;
+    }
+    else if (!strcmp(namespaceName, "System.Runtime.Intrinsics"))
+    {
+        // Vector128<T> etc
+        if (HAS_PREFIX(className, "Vector") && !strcmp(methodName, "get_IsHardwareAccelerated"))
+            return NI_IsSupported_False;
+
+        // Fall back to managed implementation for everything else.
+        return NI_Illegal;
+    }
+    else if (HAS_PREFIX(namespaceName, "System.Runtime.Intrinsics"))
+    {
+        // Architecture-specific intrinsics.
+        if (!strcmp(methodName, "get_IsSupported"))
+            return NI_IsSupported_False;
+
+        // Every intrinsic except IsSupported is PNSE in interpreted-only mode.
+        return NI_Throw_PlatformNotSupportedException;
+    }
+    else if (HAS_PREFIX(namespaceName, "System.Runtime"))
+    {
+        if (!strcmp(namespaceName, "System.Runtime.CompilerServices"))
+        {
+            if (!strcmp(className, "Unsafe"))
+            {
+                // The members of the S.R.CS.Unsafe namespace have IL generated for them elsewhere in the runtime;
+                //  we want to use that generated IL. It should Just Work once we support unsafe accessors.
+                // If the JIT is available our fallback to calling the JITted code will also work.
+                return NI_Illegal;
+            }
+            else if (!strcmp(className, "StaticsHelpers"))
```

Comment:
```suggestion
            if (!strcmp(className, "StaticsHelpers"))
```
We generate IL for number of methods too. We do not need to have them all enumerated in the interpreter.